src/main.go

```diff
@@ -0,0 +1,51 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"go/parser"
+	"go/token"
+	"log"
+	"os"
+)
+
+type logger interface {
+	log(format string, args ...interface{}) error
+}
+
+type checker interface {
+	run(fset *token.FileSet, logger logger) error
+}
+
+type consoleLogger struct{}
+
+func (c consoleLogger) log(format string, args ...interface{}) error {
+	_, err := fmt.Printf(format, args...)
+	return err
+}
+
+var checkers = []checker{}
```

Comment:
It may be better to have `checkers` as local variable inside `main`.
Or is there any particular reason for them to be global?